cogito/put.go

```diff
@@ -94,30 +94,18 @@ func NewPutter(ghAPI string, log hclog.Logger) *ProdPutter {
 	}
 }
 
-func (putter *ProdPutter) LoadConfiguration(in io.Reader, args []string) error {
-	dec := json.NewDecoder(in)
-	if err := dec.Decode(&putter.Request); err != nil {
-		return fmt.Errorf("put: parsing request: %s", err)
-	}
-	putter.Request.Env.Fill()
-
-	if err := putter.Request.Source.ValidateLog(); err != nil {
-		return fmt.Errorf("put: %s", err)
-	}
+func (putter *ProdPutter) LoadConfiguration(input []byte, args []string) error {
 	putter.log = putter.log.Named("put")
-	putter.log.SetLevel(hclog.LevelFromString(putter.Request.Source.LogLevel))
+	putter.log.Debug("started")
+	defer putter.log.Debug("finished")
 
-	putter.log.Debug("started",
-		"source", putter.Request.Source,
-		"params", putter.Request.Params,
-		"environment", putter.Request.Env,
-		"args", args)
-
-	if err := putter.Request.Source.Validate(); err != nil {
-		return fmt.Errorf("put: %s", err)
+	request, err := NewPutRequest(input)
+	if err != nil {
+		return err
 	}
-
-	putter.log.Debug("after validation and defaults",
+	putter.Request = request
+	putter.log.SetLevel(hclog.LevelFromString(request.Source.LogLevel))
```

Comment:
Why do you re-set the log level? It's already set to the value of `request.Source.LogLevel` on line 46 of `cogito/main.go`, isn't it?

Reply:
I should have added a comment indeed. I am also not 100% sure of what to do.

In main we peek at the log level but in a opaque way, to allow to go on and give more logging before failing.
If at that moment it is set to "banana", such value will be accepted and passed to the logging system, which will swallow it, recognize it as invalid and decide to set the level as INFO.

When we arrive here at line 107, the level has been validated (in Source.Validate, in this PR), so it seems it is indeed useless to re-assign. The reason why we are doing it is because of the mapping (see Source.Validate):
```go
	"silent": "off", // different
```
When we were using the ofcourse package, we published `silent` as a valid log level.
If a pipeline is using `silent`, without the re-assignment of line 107, silent would have been recognized as invalid and then INFO in main.
With this 2nd assignment, the level would be set to `off`.

So this is a good question, because it allows me to ask: is it worthwhile to go to the current confusion (well, before this PR it was even way worse) just for this theoretical backward compatibility?

I am now tempted to say no. If a pipeline really exists out there with `silent`, it will simply start to log at INFO. If the owner notices, he/she simply has to replace `silent` with `off` in the pipeline.

Another possibility that I even considered and that might be even better is to reduce the levels to simply INFO and DEBUG, for two reasons:

1. the code only uses INFO or DEBUG and will keep doing this; the other levels do not make sense.
2. I cannot see a valid case to set the logging to `off`...

Opinions?

Reply:
So, as you explained in another PR, you assign a log level before to parse the source, then you tune with what is defined in `source`.

Reply:
> Opinions?

I would just drop the `silence`/`off` support. It's not like they'll see the logs in their jobs. They'll have to go look at the resource steps to see them. If they do so, then what's the point of having turned logging off?

Reply:
This looks like a reasonable compromise indeed.

Reply:
Dropped the silence -> off support in 77d369f